contracts/utils/EnumerableSet.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.5.10;
+
+
+/**
+ * @title EnumerableSet
+ * @dev Data structure - https://en.wikipedia.org/wiki/Set_(abstract_data_type)
+ *
+ * An EnumerableSet.AddressSet is a data structure containing a number of unique addresses.
+ *
+ *  - It is possible to add and remove addresses in O(1).
+ *  - It is also possible to query if the AddressSet contains an address in O(1).
+ *  - It is possible to retrieve an array with all the addresses in the AddressSet using enumerate.
+ *    This operation is O(N) where N is the number of addresses in the AddressSet. The order in
+ *    which the addresses are retrieved is not guaranteed.
+ *
+ * Initialization of a set must include an empty array:
+ * `EnumerableSet.AddressSet set = EnumerableSet.AddressSet({values: new address[](0)});`
+ *
+ * @author Alberto Cuesta Cañada
+ */
+library EnumerableSet {
+
+    event ValueAdded(address value);
```

Comment:
If you don't provide built-in events with your other libraries I wouldn't provide them here either. It would be confusing for your users. I removed them.

In my own projects I follow the principle of always emitting an event for any state change, then I track the events in an off-chain database to avoid unnecessary calls and smart contract code. But thinking about it, the library really doesn't have a state, so maybe it is right to have the events elsewhere.